src/QMCWaveFunctions/EinsplineSetBuilderESHDF.fft.cpp

```diff
@@ -124,106 +126,119 @@ bool EinsplineSetBuilder::ReadOrbitalInfo_ESHDF(bool skipChecks)
   // Read atom orbital info from xml //
   /////////////////////////////////////
   // construct Super2Prim mapping.
-  if (Super2Prim.size() == 0)
-  {
-    //SourcePtcl->convert2Cart(SourcePtcl->R);
-    Super2Prim.resize(SourcePtcl->R.size(), -1);
-    std::vector<int> prim_atom_counts;
-    prim_atom_counts.resize(IonPos.size(), 0);
-    for (int i = 0; i < SourcePtcl->R.size(); i++)
+  // Check the number of ions in the XML and 
+  // HDF5. Skip the checks in some situations but 
+  // not others. See Issue #2224.
+
+  if (SourcePtcl->R.size() != IonPos.size()) {
+    app_error() << "The ion sizes differ between the XML and HDF5.\n";
```

Comment:
This error message should print the sizes, to make debugging issues faster  (test_einset is failing).

Reply:
I have now modified it to print out the different sizes.